src/tracks/InstrumentTrack.cpp

```diff
@@ -375,7 +413,9 @@ void InstrumentTrack::processInEvent( const MidiEvent& event, const MidiTime& ti
 			break;
 	}
 
-	if( eventHandled == false && instrument()->handleMidiEvent( event, time, offset ) == false )
+	// If the event wasn't handled, check if there's a loaded instrument and if so send the
+	// event to it. If it returns false means the instrument didn't handle the event, so we trigger a warning.
+	if( eventHandled == false && instrument() && instrument()->handleMidiEvent( event, time, offset ) == false )
```

Comment:
The event will not be handled if `instrument()` is null, so you should change the conditional.

Reply:
Nice catch! If I have `eventHandled == false && ( instrument() == nullptr || instrument()->handleMidiEvent( event, time, offset ) == false )` the call to handleMidiEvent will be skipped when instrument is null right? And just a readability question, should I write `instrument() == nullptr` or `!instrument()` there? I'm thinking the former is more readable even though a little more verbose.

Reply:
You're right. I prefer `!foo` to `foo == nullptr`, but there are no rules for comparing values with `nullptr`.
Also, there are some alternatives like `eventHandled == false && !( instrument() && instrument()->handleMidiEvent( event, time, offset ) )`.